docs/faq.rst

```diff
@@ -0,0 +1,115 @@
+
+In the pg_auto_failover design, the following two things are needed for the
+monitor to be able to orchestrate nodes integration completely:
+
+ 1. Health Checks must be successful
+
+    The monitor runs periodic health checks with all the nodes registered
+    in the system. Those *health checks* are Postgres connections from the
+    monitor to the registered Postgres nodes, and use the ``hostname`` and
+    ``port`` as registered.
+
+    The ``pg_autoctl show state`` commands column *Reachable* contains
+    "yes" when the monitor could connect to a specific node, "no" when this
+    connection failed, and "unknown" when no connection has been attempted
+    yet, since the last startup time of the monitor.
+
+    The *Reachable* column from ``pg_autoctl show state`` command output
+    must show a "yes" entry before a new standby node can be orchestrated
+    up to the "secondary" goal state.
+
+ 2. pg_autoctl service must be running
+
+    The pg_auto_failover monitor works by assigning goal states to
+    individual Postgres nodes. The monitor will not assign a new goal state
+    until the current one has been reached.
+
+    To implement a transition from the current state to the goal state
+    assigned by the monitor, the pg_autoctl service must be running on
+    every node.
+
+When your new standby node stays in the "catchingup" state for a long time,
+please check that the node is reachable from the monitor given its
+``hostname`` and ``port`` known on the monitor, and check that the
+``pg_autoctl run`` command is running for this node.
+
+The state of the system is blocked, what should I do?
+-----------------------------------------------------
+
+This question is a general case situation that is similar in nature to the
+previous situation, reached when adding a new standby to a group of Postgres
+nodes. Please check the same two elements: the monitor health cheks are
+successful, and the ``pg_autoctl run`` command is running.
+
+The monitor is a SPOF in pg_auto_failover design, how should we handle that?
+----------------------------------------------------------------------------
+
+In the current pg_auto_failover design, there can be a single monitor node.
+That is a *Single Point of Failure*. When the monitor node is not available,
+there can be no state change in the system, which means that no failover can
+happen.
+
+Losing a monitor node when the system is stable has no availability impact,
+as the Postgres streaming replication setup does not depend on the monitor
+to operate normally. If after losing the monitor node then another Postgres
+node becomes unavailable, then a failover will not happen as expected.
+
+So while the impact of losing a monitor node is limited, we still have a
+SPOF in current pg_auto_failover architecture. In the current version of
+things it is expected that a backup and recovery mechanism (PITR) be
+deployed alongside pg_auto_failover, and this should include the monitor.
```

Comment:
This section was more about what we should do about it as developers/contributors to the project, rather than how the user may cope with the situation. I have begun writing about what the users should do, and the state of things is not good. We have turned a blind eye to the problem, and I guess we are going to have to provide a better solution very soon, I'm wondering about 1.4 itself.

Reply:
It's actually not that bad but requires a restart of the Postgres services to register all the nodes again, because we need to `pg_autoctl drop node` and then `pg_autoctl create postgres` again. It might be nice to think of a better way to integrate this procedure in a way that does not have to restart Postgres.

Also, When using the `pg_autoctl create postgres` command on-top of a pre-existing setup, we should certainly re-use the full configuration file rather than insisting about the command line defaults up-front.

Reply:
These instructions are much more helpful from the user point of view!